server/src/HttpServer.hs

```diff
@@ -49,22 +66,25 @@ new core =
     when (configEnableJwtAuth $ coreConfig core) $
       middleware $ jwtMiddleware $ configJwtSecret $ coreConfig core
 
-    get (regex "^") $ do
+    get (regex "^") $ measureRequestDuration core $ do
```

Comment:
This works, but with WAI applications, there's generally a nicer way of doing this: middlewares.
With scotty, they're a bit more hidden than in a plain warp app, but you can already see one being defined above.

With that, you get access to the lower levels, below scotty's own abstractions like `ActionT` which might simplify all this a bit.

You can also look at warpmachine for an inspiration, but it's a bit more complex than what we need here: https://github.com/channable/warpmachine/blob/fa12cf21916619e14734ed656b37691a72ae7f7d/src/Warpmachine.hs#L72-L76